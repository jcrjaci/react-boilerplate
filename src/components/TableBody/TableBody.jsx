import React from 'react';
import PropTypes from 'prop-types';
import './TableBody.scss';

const TableBody = ({ data }) => (
  <div className="table-body">
    {data.map(({
      id, name, rank, symbol, price_usd: price,
    }) => (
      <div className="table-line" key={id}>
        <div className="rank">{rank}</div>
        <div className="symbol">{symbol}</div>
        <div className="name">{name}</div>
        <div className="price">{price}</div>
      </div>
    ))}
  </div>
);

TableBody.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object),
};

TableBody.defaultProps = {
  data: [],
};

export default TableBody;
