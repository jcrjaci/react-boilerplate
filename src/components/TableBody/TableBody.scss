.table-body {
  display: flex;
  flex-direction: column;

  .table-line {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }
}