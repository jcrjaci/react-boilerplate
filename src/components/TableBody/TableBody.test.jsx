import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import TableBody from './TableBody';

configure({ adapter: new Adapter() });

describe('<TableBody />', () => {
  const data = [
    { id: 1, name: 'bitcoin', symbol: 'BTC' },
    { id: 2, name: 'litecoin', symbol: 'LTC' },
  ];
  const wrapper = shallow(<TableBody data={data} />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render empty table body', () => {
    wrapper.setProps({ data: [] });

    const tableLine = wrapper.find('.table-line');
    expect(tableLine).toHaveLength(0);
  });

  it('should render 3 table line inside table body', () => {
    const newData = [
      ...data,
      { id: 3, name: 'Ethereum', symbol: 'ETH' },
    ];
    wrapper.setProps({ data: [...newData] });
    const tableLine = wrapper.find('.table-line');

    expect(tableLine).toHaveLength(3);

    expect(tableLine.at(0).key()).toBe('1');
    expect(tableLine.at(0).find('.name').text()).toBe('bitcoin');
    expect(tableLine.at(0).find('.symbol').text()).toBe('BTC');
    expect(tableLine.at(0).find('.rank').text()).toBe('');
    expect(tableLine.at(0).find('.price').text()).toBe('');

    expect(tableLine.at(1).find('.name').text()).toBe('litecoin');
    expect(tableLine.at(1).find('.symbol').text()).toBe('LTC');
    expect(tableLine.at(1).find('.rank').text()).toBe('');
    expect(tableLine.at(1).find('.price').text()).toBe('');

    expect(tableLine.at(2).find('.name').text()).toBe('Ethereum');
    expect(tableLine.at(2).find('.symbol').text()).toBe('ETH');
    expect(tableLine.at(2).find('.rank').text()).toBe('');
    expect(tableLine.at(2).find('.price').text()).toBe('');
  });
});
