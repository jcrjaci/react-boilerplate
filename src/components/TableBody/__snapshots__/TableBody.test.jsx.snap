// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableBody /> should render correctly 1`] = `
<div
  className="table-body"
>
  <div
    className="table-line"
    key="1"
  >
    <div
      className="rank"
    />
    <div
      className="symbol"
    >
      BTC
    </div>
    <div
      className="name"
    >
      bitcoin
    </div>
    <div
      className="price"
    />
  </div>
  <div
    className="table-line"
    key="2"
  >
    <div
      className="rank"
    />
    <div
      className="symbol"
    >
      LTC
    </div>
    <div
      className="name"
    >
      litecoin
    </div>
    <div
      className="price"
    />
  </div>
</div>
`;
